### PR TITLE
chore: E2e tests for iOS platform

### PR DESCRIPTION
**Pull request by rahulnr7**
1) revived tests for iOS
2) manage locators separately for iOS and Android if needed
3) added utils

### CODE REVIEW

File: apps/wallet-mobile/e2e/screens/restoreWalletFlow.screen.ts [hunk 1]
```diff
@@ -5,14 +5,17 @@ export const restore24WordWalletButton = () => element(by.id('restore24WordWalle
...
 export const mnemonicRestoreWalletButton = () => element(by.id('restoreButton'))
 
 export const walletChecksumText = () => element(by.id('walletChecksum'))
 export const verifyWalletContinueButton = () => element(by.id('verifyWalletContinueButton'))
 
 export const credentialsView = () => element(by.id('credentialsView'))
-export const walletNameInput = () =>  element(by.type(`android.widget.EditText`)).atIndex(0)
-export const spendingPasswordInput = () => element(by.type(`android.widget.EditText`)).atIndex(1)
-export const repeatSpendingPasswordInput = () => element(by.type(`android.widget.EditText`)).atIndex(2)
+// export const walletNameInput = () =>  element(by.type(`android.widget.EditText`)).atIndex(0)
```
Comment (Nebyt) [inline apps/wallet-mobile/e2e/screens/restoreWalletFlow.screen.ts]: Please remove these commented lines

File: apps/wallet-mobile/e2e/screens/restoreWalletFlow.screen.ts [hunk 2]
```diff
@@ -5,14 +5,17 @@ export const restore24WordWalletButton = () => element(by.id('restore24WordWalle
...
+export const walletNameInput = () => element(by.id('walletNameInput'))
+// export const spendingPasswordInput = () => element(by.type(`android.widget.EditText`)).atIndex(1)
+// export const repeatSpendingPasswordInput = () => element(by.type(`android.widget.EditText`)).atIndex(2)
```
Comment (Nebyt) [inline apps/wallet-mobile/e2e/screens/restoreWalletFlow.screen.ts]: Please remove these commented lines

File: apps/wallet-mobile/e2e/tests/_ios/create-wallet.test.ts [hunk 3]
```diff
@@ -1,31 +1,30 @@
...
         await expect(createWalletFlow.credentialsView()).toBeVisible();
         await createWalletFlow.walletNameInput().typeText(WALLET_NAME)
-        await device.disableSynchronization()
-        await createWalletFlow.spendingPasswordInput().typeText(SPENDING_PASSWORD)
-        await createWalletFlow.repeatSpendingPasswordInput().typeText(`${SPENDING_PASSWORD}\n`)
-        await device.enableSynchronization()
+   
+        await createWalletFlow.spendingPasswordInput().tap()
+        await createWalletFlow.spendingPasswordInput().typeText(`${SPENDING_PASSWORD}\n`)
+       // await device.enableSynchronization()
```
Comment (Nebyt) [inline apps/wallet-mobile/e2e/tests/_ios/create-wallet.test.ts]: Please remove these lines if we don't need it

File: apps/wallet-mobile/e2e/tests/_ios/create-wallet.test.ts [hunk 4]
```diff
@@ -1,31 +1,30 @@
...
+        await waitFor(createWalletFlow.repeatSpendingPasswordInput()).toBeVisible().withTimeout(10000)
+        await createWalletFlow.repeatSpendingPasswordInput().tap()
+        await createWalletFlow.repeatSpendingPasswordInput().typeText(SPENDING_PASSWORD)
+       // await device.enableSynchronization()
```
Comment (Nebyt) [inline apps/wallet-mobile/e2e/tests/_ios/create-wallet.test.ts]: Please remove these lines if we don't need it

**Review by Nebyt — CHANGES_REQUESTED** [review]

File: apps/wallet-mobile/e2e/screens/restoreWalletFlow.screen.ts (hunk 2 above)
Comment (rahulnr7) [reply apps/wallet-mobile/e2e/screens/restoreWalletFlow.screen.ts]: done. thanks

File: apps/wallet-mobile/e2e/tests/_ios/create-wallet.test.ts (hunk 4 above)
Comment (rahulnr7) [reply apps/wallet-mobile/e2e/tests/_ios/create-wallet.test.ts]: done thanks

File: apps/wallet-mobile/e2e/tests/_ios/create-wallet.test.ts (hunk 3 above)
Comment (rahulnr7) [reply apps/wallet-mobile/e2e/tests/_ios/create-wallet.test.ts]: done

File: apps/wallet-mobile/package.json [hunk 5]
```diff
@@ -31,7 +43,7 @@
     "run:android:dev:debug": "ENVFILE=.env react-native run-android --mode=devDebug --appIdSuffix=dev --verbose",
     "run:android:nightly:debug": "ENVFILE=.env.nightly react-native run-android --mode=nightlyDebug --appIdSuffix=nightly --verbose",
     "run:android:nightly:release": "ENVFILE=.env.nightly react-native run-android --mode=nightlyRelease --appIdSuffix=nightly --verbose",
-    "run:ios:dev:debug": "ENVFILE=.env react-native run-ios --mode=Debug --scheme=yoroi --verbose",
+    "run:ios:yoroi:debug": "ENVFILE=.env react-native run-ios --mode=Debug --scheme=yoroi --verbose",
```
Comment (Nebyt) [inline apps/wallet-mobile/package.json]: I don't think we need to do that

**Review by Nebyt — APPROVED** [review]

File: apps/wallet-mobile/package.json (hunk 5 above)
Comment (rahulnr7) [reply apps/wallet-mobile/package.json]: Yeah, you are right. Reverted

**Pull request merged**